opentelemetry-sdk/tests/metrics/export/test_export.py

```diff
@@ -34,10 +40,210 @@ def test_export(self):
         )
         kvp = {"environment": "staging"}
         label_set = meter.get_label_set(kvp)
-        handle = metric.get_handle(label_set)
-        result = '{}(data="{}", label_values="{}", metric_data={})'.format(
-            ConsoleMetricsExporter.__name__, metric, label_set, handle
+        aggregator = CounterAggregator()
+        record = MetricRecord(aggregator, label_set, metric)
+        result = '{}(data="{}", label_set="{}", value={})'.format(
+            ConsoleMetricsExporter.__name__,
+            metric,
+            label_set.labels,
+            aggregator.check_point,
         )
         with mock.patch("sys.stdout") as mock_stdout:
-            exporter.export([(metric, label_set)])
+            exporter.export([record])
             mock_stdout.write.assert_any_call(result)
+
+
+class TestBatcher(unittest.TestCase):
+    def test_aggregator_for_counter(self):
+        batcher = UngroupedBatcher(True)
+        self.assertTrue(
+            isinstance(
+                batcher.aggregator_for(metrics.Counter), CounterAggregator
+            )
+        )
+
+    # TODO: Add other aggregator tests
+
+    def test_check_point_set(self):
+        meter = metrics.Meter()
+        batcher = UngroupedBatcher(True)
+        aggregator = CounterAggregator()
+        metric = metrics.Counter(
+            "available memory",
+            "available memory",
+            "bytes",
+            int,
+            meter,
+            ("environment",),
+        )
+        aggregator.update(1.0)
+        label_set = {}
+        batch_map = {}
+        batch_map[(metric, "")] = (aggregator, label_set)
```

Comment:
it looks like this block sets a very specific expectation around the format of the batch_map. In general, I feel like this is a bit error-prone as assigning the value requires knowing the right pair of tuples to set as the key and the value.

Is there some refactoring that can be done to not require direct manipulation of the batch_map, providing helper methods instead? that way, there's clear documentation on the structure of the key-value pairs, or hopefully one doesn't have to set the key-value pairs at all.

Reply:
Good point. I think adding a helper method to assign values to the batch map would be good.

Reply:
EDIT: There seems to be some problems with circular imports (since __init__.py depends on the batcher.py and batcher.py would have to depend on __init__.py for typing). Sphinx also complains too since the forward refs aren't able to be found. Any ideas on how to fix this?

Reply:
where do you see the circular imports? In general this can be solved by factoring out the circular component in it's own module, and having the circular importees import that instead.